core/src/main/scala/besom/internal/ResourceOptions.scala

```diff
@@ -44,13 +46,25 @@ sealed trait ResourceOptions:
   def ignoreChanges: List[String]
   def replaceOnChanges: List[String]
   def retainOnDelete: Boolean
+  def urn: Option[URN]
+
+  private[besom] def hasURN: Boolean = urn.isDefined
+  private[besom] def hasImportId(using Context): Boolean = this match
+    case cr: CustomResourceOptions         => cr.importId.isDefined
+    case sr: StackReferenceResourceOptions => sr.importId.isDefined
+    case _                                 => false
+
+  private[besom] def getImportId(using Context): Option[ResourceId] = this match
+    case cr: CustomResourceOptions         => cr.importId
+    case sr: StackReferenceResourceOptions => sr.importId
+    case _                                 => None
 
 final case class CustomResourceOptions private[internal] (
   common: CommonResourceOptions,
   provider: Option[ProviderResource],
   deleteBeforeReplace: Boolean,
   additionalSecretOutputs: List[String],
-  importId: Option[NonEmptyString] // TODO should this be Id?
+  importId: Option[ResourceId] // TODO should this be Id?
```

Comment:
nit: the todo note appears to be fixed now :)

Reply:
```suggestion
  importId: Option[ResourceId]
```

Reply:
it isn't really, resource options require a rewrite